go/vt/vtctl/workflow/server.go

```diff
@@ -669,23 +738,119 @@ func (s *Server) LookupVindexExternalize(ctx context.Context, req *vtctldatapb.L
 	resp := &vtctldatapb.LookupVindexExternalizeResponse{}
 
 	if vindex.Owner != "" {
-		// If there is an owner, we have to delete the streams. Once we externalize it
+		// If there is an owner, we have to stop the streams. Once we externalize it
 		// the VTGate will now be responsible for keeping the lookup table up to date
 		// with the owner table.
-		if _, derr := s.WorkflowDelete(ctx, &vtctldatapb.WorkflowDeleteRequest{
-			Keyspace:         req.TableKeyspace,
-			Workflow:         req.Name,
-			KeepData:         true, // Not relevant
-			KeepRoutingRules: true, // Not relevant
-		}); derr != nil {
-			return nil, vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "failed to delete workflow %s: %v", req.Name, derr)
+		err = forAllShards(targetShards, func(si *topo.ShardInfo) error {
+			tabletInfo, err := s.ts.GetTablet(ctx, si.PrimaryAlias)
+			if err != nil {
+				return err
+			}
+			// Stop the workflow.
+			_, err = s.tmc.UpdateVReplicationWorkflow(ctx, tabletInfo.Tablet, &tabletmanagerdatapb.UpdateVReplicationWorkflowRequest{
+				Workflow: req.Name,
+				State:    ptr.Of(binlogdatapb.VReplicationWorkflowState_Stopped),
+			})
+			if err != nil {
+				return vterrors.Wrapf(err, "failed to stop workflow %s on shard %s/%s", req.Name, tabletInfo.Keyspace, tabletInfo.Shard)
+			}
+			// Mark workflow as frozen.
+			query := fmt.Sprintf(SqlFreezeWorkflow, Frozen,
+				encodeString(tabletInfo.DbName()), encodeString(req.Name))
+			_, err = s.tmc.VReplicationExec(ctx, tabletInfo.Tablet, query)
```

Comment:
I think that we should update the message in the `UpdateVReplicationWorkflow` call above, which means we'll have to add support for that to the RPC as an optional proto field.

Reply:
Done, thanks for this! This is much cleaner now.